### PR TITLE
feat(dedicated): delete subnet from vrack ipv6

### PR DESCRIPTION
**Pull request by mhelhali-soufien**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/additional-ipv6-feature`
| Bug fix?         |no
| New feature?     | yes
| Breaking change? |no
| Tickets          | Fix #MANAGER-14042
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/modules/vrack/src/vrack-services/ip/add/controller.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export default class IPv6SubnetAddModalCtrl {
+  onSubmit() {
+    if (this.formName.$valid) {
+      this.onConfirm()(this.subnet, this.address);
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-services/ip/add/controller.js:4]: you confuse `&` and `=` bindings. That's why you have a weird call functions here.
As you're using `&` bindings, you should set `data-on-confirm=myFunction(this.subnet, this.address)` directly in the modal html declaration

File: packages/manager/modules/vrack/src/vrack-services/ip/delete/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  .component('deleteIpv6SubnetModal', {
+    template,
+    controller: [
+      '$scope',
+      function() {
+        this.cancel = () => {
+          this.isOpenModal = false;
+        };
+      },
+    ],
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-services/ip/delete/index.js:18]: `$scope` injection is useless here.
Moreover, it's weird to have the close function defined here but not the open one.
As you're using an `onConfirm` function, sounds more logic to have also `onCancel` function

File: packages/manager/modules/vrack/src/vrack-services/ip/ip.controller.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  $onInit() {
+    const parentCallBack = this.refreshData();
+    this.vrackAssignedIpService
+      .getIpVrackSubnet(this.serviceName, this.ip.niceName)
+      .then(({ data }) => {
+        this.subnets = data;
+      });
+
+    this.openAddSubnetModal = () => {
+      this.openAddModal()((subnet, address) => {
```
Comment (qpavy) [inline packages/manager/modules/vrack/src/vrack-services/ip/ip.controller.js:21]: Same as https://github.com/ovh/manager/pull/11538/files#r1580801446

File: packages/manager/modules/vrack/src/vrack-services/ip/ip.controller.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                this.$translate.instant('vrack_error'),
+                (err.data && err.data.message) || err.message || '',
+              ].join(' '),
+            );
+          });
+      });
+    };
+
+    this.openDeleteSubnetModal = (subnet) => {
+      this.openDeleteModal()(() => {
```
Comment (qpavy) [inline packages/manager/modules/vrack/src/vrack-services/ip/ip.controller.js:43]: Same as https://github.com/ovh/manager/pull/11538/files#r1580801446

**Review by qpavy — CHANGES_REQUESTED** [review]

File: packages/manager/modules/vrack/src/vrack-services/ip/ip.less [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  &__assigned-ip {
+    &__sub-item {
+      display: flex;
+      justify-content: end;
+      flex-flow: wrap;
+    }
+
+    &__subnet {
+      list-style: none;
+      padding: 0.625rem 0.9375rem;
```
[the remark this answers is not in the document]
Comment (mhelhali-soufien) [reply packages/manager/modules/vrack/src/vrack-services/ip/ip.less:13]: It should be like as ipv4

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 6]
```diff
@@ -171,24 +172,15 @@ export default class VrackMoveDialogCtrl {
...
-
     this.$scope.$on('$destroy', () => {
       if (this.poller) {
         this.$timeout.cancel(this.poller);
       }
     });
+
+    this.isIPv6 = (service) => {
+      return !!service.id.match(/::/);
+    };
```
Comment (darsene) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:183]: try to keep on way for declaring function

```suggestion
     static isIPv6(service) {
      return !!service.id.match(/::/);
    };
```

File: packages/manager/modules/vrack/src/dashboard/vrack.service.js [hunk 7]
```diff
@@ -6,4 +6,10 @@ export default class Vrack {
   addIpv6(serviceName, ipv6) {
     return this.$http.post(`/vrack/${serviceName}/ipv6`, { block: ipv6 });
   }
+
+  deleteIpv6(serviceName, ipv6) {
+    return this.$http.delete(
+      `/vrack/${serviceName}/ipv6/${window.encodeURIComponent(ipv6)}`,
```
Comment (darsene) [inline packages/manager/modules/vrack/src/dashboard/vrack.service.js:12]:
```suggestion
      `/vrack/${serviceName}/ipv6/${encodeURIComponent(ipv6)}`,
```

File: packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+            );
+          });
+      },
+      onCancel: () => {
+        this.deleteSubnetModalContext.isOpenModal = false;
+      },
+    };
+  }
+
+  refrechData() {
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js:90]:
```suggestion
  refreshData() {
```

File: packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        taskId,
+      })
+      .$promise.then(() => {
+        this.$timeout(() => {
+          this.watingTask(taskId);
+        }, POLLING_INTERVAL);
+      })
+      .catch(() => {
+        this.loader = false;
+        this.refrechData();
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js:112]:
```suggestion
        this.refreshData();
```

File: packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.service.js [hunk 10]
```diff
@@ -23,4 +23,14 @@ export default class VrackAssignedIpService {
...
   }
+
+  deleteIpVrackSubnet(serviceName, ipBlock, routedSubrange) {
+    return this.$http.delete(
+      `/vrack/${window.encodeURIComponent(
+        serviceName,
+      )}/ipv6/${window.encodeURIComponent(
+        ipBlock,
+      )}/routedSubrange/${window.encodeURIComponent(routedSubrange)}`,
+    );
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.service.js:34]:
```suggestion
    return this.$http.delete(
      `/vrack/${encodeURIComponent(
        serviceName,
      )}/ipv6/${encodeURIComponent(
        ipBlock,
      )}/routedSubrange/${encodeURIComponent(routedSubrange)}`,
    );
```

File: packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+            nexthop,
+          })
+          .then(({ data }) => {
+            this.watingTask(data.id);
+          })
+          .catch((err) => {
+            this.CucCloudMessage.error(
+              [
+                this.$translate.instant('vrack_error'),
+                (err.data && err.data.message) || err.message || '',
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js:53]:
```suggestion
                error.data?.message || err.message || '',
```

File: packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+        this.vrackAssignedIpv6Service
+          .deleteIpVrackSubnet(this.serviceName, this.ip.niceName, subnet)
+          .then(({ data }) => {
+            this.watingTask(data.id);
+          })
+          .catch((err) => {
+            this.CucCloudMessage.error(
+              [
+                this.$translate.instant('vrack_error'),
+                (err.data && err.data.message) || err.message || '',
```
Comment (darsene) [inline packages/manager/modules/vrack/src/vrack-associated-services/ipv6/ipv6.controller.js:79]:
```suggestion
               error.data?.message || err.message || '',
```

**Review by darsene — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Pull request merged**